index.js

```diff
@@ -15,10 +17,57 @@ winston.level = process.env.WINSTON_LEVEL;
 app.use(bodyParser.urlencoded({ extended: true }));
 app.use(bodyParser.json());
 
+app.get('/login', function(req, res) {
+  winston.log("info", "login");
+  var apiKey = process.env.BB_KEY;
+  var apiSecret = process.env.BB_SECRET;
+
+  if(typeof apiKey !== 'undefined' || typeof apiSecret !== 'undefined') {
+    var oauth = new OAuth.OAuth(
+      'https://bitbucket.org/api/1.0/oauth/request_token',
+      'https://bitbucket.org/!api/1.0/oauth/access_token',
+      apiKey,
+      apiSecret,
+      '1.0',
+      'https://git-mirror-sync.herokuapp.com/bbauth',
+      'HMAC-SHA1'
+    );
+
+    oauth.getOAuthRequestToken(function(error, oauth_token, oauth_token_secret, results) {
+      if (error) {
+        winston.log("error", error);
+        res.sendStatus("500");
+      } else {
+        winston.log("debug", oauth_token);
+        winston.log("debug", oauth_token_secret);
+
+        res.redirect("https://bitbucket.org/api/1.0/oauth/authenticate?oauth_token="+ oauth_token);
+      }
+    });
+  } else {
+    winston.error("missing bitbucket client_id expected enviroment variable BB_KEY");
+    res.sendStatus(500);
+  }
+});
+
+app.get('/bbauth', function(req, res) {
+  winston.log("info", "GET /bbauth");
+
+  var auth = req.query.oauth_token;
+  var verifier = req.query.oauth_verifier;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['verifier' is defined but never used.](https://www.codacy.com/app/jeffhann/git-mirror-sync/file/757207656/issues/source?bid=2446500&fileBranchId=2449083#l57)